src/hotspot/cpu/x86/c1_Runtime1_x86.cpp

```diff
@@ -1111,7 +1111,17 @@ OopMapSet* Runtime1::generate_code_for(StubID id, StubAssembler* sasm) {
       }
 
       break;
-
+#ifdef _LP64
+    case load_klass_id:
+      {
+        StubFrame f(sasm, "load_klass", dont_gc_arguments);
+        sasm->save_live_registers_no_oop_map(true);
```

Comment:
I presume calling `*_no_oop_map` here is intentional, and valid for `call_VM_leaf`? I.e. no safepoint expected here?

Reply:
Yes exactly, leaf calls don't require an oop map.

Reply:
And no, no safepoints expected.